src/app/components/NavBar/index.tsx

```diff
@@ -1,20 +1,49 @@
-import { FC, useState } from "react";
+import { FC, useEffect, useRef, useState } from "react";
 
 import { Bars3Icon } from "@heroicons/react/24/outline";
 import DesktopNavBar from "./DesktopNavBar";
 import Image from "next/image";
 import Link from "next/link";
 import MobileNavBar from "./MobileNavBar";
 import { useScrollColor } from "./hooks/useScrollColor";
+import useWindowSize from "./hooks/useWindowSize";
 
 const NavBar: FC = () => {
   const { bgColor, textColor } = useScrollColor();
+  const { width } = useWindowSize();
+  const headerRef = useRef<HTMLDivElement | null>(null);
   const [mobileMenuOpen, setMobileMenuOpen] = useState(false);
 
+  useEffect(() => {
+    let prevPosY = window.scrollY;
+
+    const handleScrollDirection = () => {
+      const headerEl = headerRef.current;
+      const currPosY = window.scrollY;
+      const direction = currPosY - prevPosY > 0 ? "down" : "up";
+      if (!headerEl) {
+        return;
+      }
+      if (direction === "down" && currPosY - prevPosY > 10 && width < 768) {
```

Comment:
on va essayez à partir de maintenant de ne pas utiliser ce qu'on appelle les [magic numbers](https://www.inf.unibz.it/~calvanese/teaching/05-06-ip/lecture-notes/uni04/node17.html).

Je parle du 10 et du 768.